common/lib/src/realm_types.dart

```diff
@@ -0,0 +1,136 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:collection';
+import 'dart:ffi';
+import 'dart:typed_data';
+
+enum RealmPropertyType {
+  int,
+  bool,
+  string,
+  // ignore: unused_field, constant_identifier_names
+  _3,
+  binary,
+  // ignore: unused_field, constant_identifier_names
+  _5,
+  mixed,
+  // ignore: unused_field, constant_identifier_names
+  _7,
+  timestamp,
+  float,
+  double,
+  decimal128,
+  object,
+  // ignore: unused_field, constant_identifier_names
+  _13,
+  linkingObjects,
+  objectid,
+  // ignore: unused_field, constant_identifier_names
+  _16,
+  uuid,
+}
+
+enum RealmCollectionType {
+  none,
+  list,
+  set,
+  dictionary,
+}
+
+class Uuid {} // TODO!
+
+class ObjectId {} // TODO!
+
+class Decimal128 {} // TODO!
```

Comment:
Comment out all types that are not supported.

Reply:
I would like to keep them in, but let generator complain. That way I can leave the support in, but still not allow the user to use them

Reply:
I think this is unnecessary confusion to developers. We are exporting all these types from the package. If you would like to keep them in the generator at least we should hide them from the exports in realm_class

Reply:
below I have added comment to hide these from the package

Reply:
Fixed